AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -141,29 +141,29 @@ + (BOOL)openURL:(NSURL *)url {
 }
 
 + (void)login {
-
-  // TODO protect with canBeUsed.
-  [[MSIdentity sharedInstance] login];
+  @synchronized(lock) {
+    if ([[MSIdentity sharedInstance] canBeUsed]) {
+      [[MSIdentity sharedInstance] login];
+    }
+  }
 }
 
 - (void)login {
-  @synchronized(lock) {
-    if (self.clientApplication == nil && self.identityConfig == nil) {
-      self.loginDelayed = YES;
-      return;
-    }
-    self.loginDelayed = NO;
-    [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
-                                  completionBlock:^(MSALResult *result, NSError *e) {
-                                    // TODO: synchronize accessToken assignment if it has threading issues
-                                    if (e) {
-                                      MSLogError([MSIdentity logTag], @"Couldn't initialize authentication client. Error: %@", e);
-                                    } else {
-                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
-                                      self.accessToken = result.accessToken;
-                                    }
-                                  }];
+  if (self.clientApplication == nil || self.identityConfig == nil) {
+    self.loginDelayed = YES;
+    return;
   }
+  self.loginDelayed = NO;
+  [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                completionBlock:^(MSALResult *result, NSError *e) {
+                                  // TODO: synchronize accessToken assignment if it has threading issues
```

Comment:
This TODO should be addressed with this PR.

Reply:
Remove the TODO if unnecessary, or handle possible threading issue here.

Reply:
After some testing, I didn't find any issues so I removed this TODO. But let me double check.

Reply:
@roland-chernov-akvelon please make sure the code is thread safe regardless of whether a race condition can actually be reproduced. Even the no-repro race conditions must be taken into account.

Reply:
I double checked and made some changes. I think it is good now.